lib/DB/TestUtils.pm

```diff
@@ -52,11 +54,11 @@ sub buildHash ($input, $config) {
 			} else {
 				$output->{$field}->{$subfield} = $input->{$key} if defined($input->{$key});
 			}
-		} elsif (grep {/^$key$/} @{ $config->{boolean_fields} }) {
+		} elsif (grep { $_ eq $key } @{ $config->{boolean_fields} }) {
 			$output->{$key} = defined($input->{$key}) && int($input->{$key}) ? true : false;
-		} elsif (grep {/^$key$/} @{ $config->{non_neg_int_fields} }) {
+		} elsif (grep { $_ eq $key } @{ $config->{non_neg_int_fields} }) {
 			$output->{$key} = int($input->{$key}) if defined($input->{$key});
-		} elsif (grep {/^$key$/} @{ $config->{non_neg_float_fields} }) {
+		} elsif (grep { $_ eq $key } @{ $config->{non_neg_float_fields} }) {
 			$output->{$key} = 0 + $input->{$key} if defined($input->{$key});
 		} else {
 			$output->{$key} = $input->{$key};
```

Comment:
This file has bugged me for a while.  Why is this part of the main `lib`?  It would be in the `t` directory.

Reply:
I can move this.